moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java

```diff
@@ -173,6 +173,54 @@
     }
   }
 
+  @Test public void emptyAdapters() throws Exception {
+    try {
+      Moshi moshi = new Moshi.Builder()
+              .add(new EmptyJsonAdapter()).build();
+    } catch (IllegalArgumentException expected){
+      assertThat(expected.getMessage()).contains(
+              "Expected at least one @ToJson or @FromJson method on", "EmptyJsonAdapter");
+    }
+  }
+
+  static class EmptyJsonAdapter{}
```

Comment:
We follow these:
https://google.github.io/styleguide/javaguide.html
https://github.com/square/java-code-styles